src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl_ModelComputation.cs

```diff
@@ -0,0 +1,257 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    internal partial class DocumentOutlineControl
+    {
+        private IWpfTextView? GetLastActiveIWpfTextView()
+        {
+            ThreadingContext.ThrowIfNotOnUIThread();
+
+            if (ErrorHandler.Failed(CodeWindow.GetLastActiveView(out var textView)))
+                return null;
+
+            return EditorAdaptersFactoryService.GetWpfTextView(textView);
+        }
+
+        /// <summary>
+        /// Starts a new task to compute the UI model.
+        /// </summary>
+        private void StartComputeUIModelTask()
+        {
+            // 'true' value is unused.  this just signals to the queue that we have work to do.
+            _computeUIModelQueue.AddWork(true);
+        }
+
+        /// <summary>
+        /// Creates the UI model.
+        /// </summary>
+        private async ValueTask<DocumentSymbolModel?> ComputeUIModelAsync(ImmutableSegmentedList<bool> unused, CancellationToken cancellationToken)
```

Comment:
nit: 
```suggestion
        private async ValueTask<DocumentSymbolModel?> ComputeUIModelAsync(ImmutableSegmentedList<bool> _, CancellationToken cancellationToken)
```

(both are fine).